### PR TITLE
ENTESB-4109 Type Core Filter does not propagate its value to query st…

### PR DESCRIPTION
**Pull request by dvirgiln**
https://issues.jboss.org/browse/ENTESB-4109

### CODE REVIEW

File: ui/src/main/java/org/artificer/ui/client/shared/services/IArtifactSearchService.java [hunk 1]
```diff
@@ -58,4 +61,17 @@
...
+     * Generate a query string using the given filters.
+     *
+     * Note: Theoretically, query generation could be client-side. However,
+     * numerous conveniences are used (Calendar, etc.) that are not supported by
+     * GWT emulation. So, for now, offloading to the server-side services.
+     *
+     * @throws org.artificer.ui.client.shared.exceptions.ArtificerUiException
+     */
+    @POST
+    @Path("query")
```
Comment (EricWittmann) [inline ui/src/main/java/org/artificer/ui/client/shared/services/IArtifactSearchService.java]: Is this path correct?  "query" returns the types?

File: ui/src/main/java/org/artificer/ui/server/services/ArtifactSearchService.java [hunk 2]
```diff
@@ -249,5 +251,25 @@ protected void zeroOutTime(Calendar cal) {
...
     }
 
+    @Override
+    public List<ArtifactTypeBean> types() throws ArtificerUiException {
+        List<String> types = null;
+        try {
+            types = ArtificerApiClientAccessor.getClient().getTypes();
+        } catch (ArtificerClientException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
```
Comment (EricWittmann) [inline ui/src/main/java/org/artificer/ui/server/services/ArtifactSearchService.java]: Need something better here?

File: ui/src/main/java/org/artificer/ui/server/services/ArtifactSearchService.java [hunk 3]
```diff
@@ -249,5 +251,25 @@ protected void zeroOutTime(Calendar cal) {
...
+        } catch (ArtificerServerException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
```
Comment (EricWittmann) [inline ui/src/main/java/org/artificer/ui/server/services/ArtifactSearchService.java]: Need something better here?

File: ui/src/main/java/org/artificer/ui/client/shared/services/IArtifactSearchService.java (hunk 1 above)
Comment (dvirgiln) [inline ui/src/main/java/org/artificer/ui/client/shared/services/IArtifactSearchService.java]: Corrected! Thanks

**Pull request merged**